virtManager/details/snapshots.py

```diff
@@ -234,7 +302,7 @@ def _get_screenshot_data_for_save(self):
     def _do_create_snapshot(self, asyncjob, xml, name, mime, sndata):
         ignore = asyncjob
 
-        self.vm.create_snapshot(xml, diskOnly=False)
+        self.vm.create_snapshot(xml, diskOnly=self._diskOnly)
```

Comment:
There's a little bug here in that self._diskOnly=True can be carried over, if user creates a disk only external snapshot, and then next creates an internal snapshot. Drop `self._diskOnly` and just calculate it _create_new_snapshot and pass it to _do_create_snapshot.

Or adjust domain.py to take the xmlobj directly, and calculate it there, so this file doesn't need to care about the snapshot flags distinction.

Reply:
Nice catch, will fix it!